verticapy/tests_new/performance/vertica/test_qprof.py

```diff
@@ -0,0 +1,1627 @@
+
+    return div
+
+
+def _compare_pandas(l_df: pd.DataFrame, r_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    function to compare two pandas dataframes
+    """
+    res = l_df.compare(
+        r_df,
+        result_names=(
+            "left",
+            "right",
+        ),
+    )
+    return res
+
+
+class TestQueryProfiler:
+    """
+    test class for QueryProfiler
+    """
+
+    _target_tables = [
+        "dc_requests_issued",
+        "dc_query_executions",
+        "dc_explain_plans",
+        "execution_engine_profiles",
+        "query_plan_profiles",
+        "query_profiles",
+        "resource_pool_status",
+        "host_resources",
+    ]
+
+    @pytest.mark.parametrize(
+        "test_name, transactions, key_id, resource_pool, target_schema, overwrite, add_profile, check_tables",
+        [
+            ("transactions", "integer", None, None, None, False, False, True),
+            ("transactions", "list_of_integers", None, None, None, False, False, True),
+            ("transactions", "tuple", None, None, None, False, False, True),
+            ("transactions", "list_of_tuples", None, None, None, False, False, True),
+            ("transactions", "single_sql", None, None, None, False, False, True),
+            ("transactions", "multiple_sql", None, None, None, False, False, True),
+            ("target_schema", "tuple", None, None, "qprof_test", False, False, True),
+            ("key_id", "tuple", "qprof_key_id", None, None, True, False, True),
+            # need to check on this
+            # ("key_id", "tuple", "qprof_key_id", None, "qprof_test", True, False, True),
+            # need to check on this
+            # ("resource_pool", "tuple", None, "aaa", "qprof_test", False, False, True),
+            (
+                "target_schema",
+                "tuple",
+                None,
+                None,
+                {"v_internal": "qprof_test", "v_monitor": "qprof_test"},
+                False,
+                False,
+                True,
+            ),
+            (
+                "target_schema",
+                "tuple",
+                None,
+                None,
+                {"v_internal": "qprof_test1", "v_monitor": "qprof_test2"},
+                False,
+                False,
+                True,
+            ),
+            # need to check on this
+            # ("overwrite", "tuple", "ut_key_12", None, "qprof_test", False, False, True),
+            # overwrite failing. should overwrite.
+            # ("overwrite", "tuple", "ut_key_12", None, "qprof_test", True, False, True),
+            ("add_profile", "single_sql", None, None, None, False, True, True),
+            # need to check. how this would be tested
+            ("check_tables", "integer", None, None, None, False, False, True),
+            # need to check. how this would be tested
+            ("check_tables", "integer", None, None, None, False, False, False),
+        ],
+    )
+    def test_query_profiler(
```

Comment:
Is this testing the whole profiler, or just the constructor?

Reply:
As mentioned in doc string, This function test QueryProfiler class and its parameters.